src/gwc/src/main/java/org/geoserver/gwc/GWC.java

```diff
@@ -319,6 +323,9 @@ public Catalog getCatalog() {
     }
 
     public GWCConfig getConfig() {
+        if (gsEnvironment != null && gsEnvironment.isStale()) {
```

Comment:
This bloc of code is repeated many times, should be centralized in a single method.